src/main/java/picard/illumina/CollectIlluminaLaneMetrics.java

```diff
@@ -84,10 +84,28 @@
     @Option(doc = ReadStructure.PARAMETER_DOC, shortName = "RS")
     public ReadStructure READ_STRUCTURE;
 
+    @Option(doc = "Include skips and molecular barcodes for both phasing and pre-phasing values.  Use this if the either of the former are in-line in the read(s) (not in index reads).")
+    public boolean INCLUDE_SKIPS_AND_MOLECULAR_BARCODES = false;
+
     @Override
     protected int doWork() {
         final MetricsFile<MetricBase, Comparable<?>> laneMetricsFile = this.getMetricsFile();
         final MetricsFile<MetricBase, Comparable<?>> phasingMetricsFile = this.getMetricsFile();
+
+        // If skips or molecular identifiers are part of the read structure and also part of the non-index reads (line),
+        // then convert them template descriptors for our purposes.
+        if (INCLUDE_SKIPS_AND_MOLECULAR_BARCODES) {
+            final List<ReadDescriptor> descriptors = READ_STRUCTURE.descriptors.stream().map(descriptor -> {
+                if (descriptor.type == ReadType.Skip || descriptor.type == ReadType.MolecularIndex) {
+                    return new ReadDescriptor(descriptor.length, ReadType.Template);
```

Comment:
in my manual tests (modifying the provider in IlluminaLaneMetricsCollectorTest.java) I don't find that having 6T70T instead of  6M70M avoids the error...so I don't see how this will work. (a test would prove me wrong...)